server/data/restClient.ts

```diff
@@ -9,21 +9,20 @@ import type { ApiConfig } from '../config'
 import type { UnsanitisedError } from '../sanitisedError'
 import { restClientMetricsMiddleware } from './restClientMetricsMiddleware'
 
-interface GetRequest {
+interface BaseRequest {
   path?: string
   query?: object | string
   headers?: Record<string, string>
   responseType?: string
   raw?: boolean
+}
+
+interface Request extends BaseRequest {
```

Comment:
Again, I'm probably missing something but could this `handle404` property be in the `BaseRequest` class and simplify the class hierarchy?